Xamarin.Forms.Xaml/MarkupExtensions/TypeExtension.cs

```diff
@@ -10,13 +10,14 @@ public class TypeExtension : IMarkupExtension<Type>
 
 		public Type ProvideValue(IServiceProvider serviceProvider)
 		{
-			if (string.IsNullOrEmpty(TypeName))
-				throw new InvalidOperationException("TypeName isn't set.");
 			if (serviceProvider == null)
 				throw new ArgumentNullException(nameof(serviceProvider));
-			var typeResolver = serviceProvider.GetService(typeof (IXamlTypeResolver)) as IXamlTypeResolver;
-			if (typeResolver == null)
+			if (!(serviceProvider.GetService(typeof(IXamlTypeResolver)) is IXamlTypeResolver typeResolver))
 				throw new ArgumentException("No IXamlTypeResolver in IServiceProvider");
+			if (string.IsNullOrEmpty(TypeName)) {
+				var li = (serviceProvider.GetService(typeof(IXmlLineInfoProvider)) is IXmlLineInfoProvider lip) ? lip.XmlLineInfo : new XmlLineInfo();
```

Comment:
This feature repeats 3 times in this PR (6 times in Xaml project). Perhaps to allocate a separate static method?
```C#
XmlLineInfo GetLineInfo(IServiceProvider serviceProvider) 
=> (serviceProvider.GetService(typeof(IXmlLineInfoProvider)) is IXmlLineInfoProvider lip) ? lip.XmlLineInfo : new XmlLineInfo()
```

Reply:
rationalizing that usage is part of #4582.